OIPA/iati/parser/IATI_2_03.py

```diff
@@ -4882,6 +4883,69 @@ def iati_activities__iati_activity__tag__narrative(self, element):
         activity_tag = self.get_model('ActivityTag')
         self.add_narrative(element, activity_tag)
 
+    def iati_activities__iati_activity__fss(self, element):
+        """"(optional) <fss> element inside <iati_activities/iati-activity>
+        element in 2.03
+        """
+        # we need to check if this element occurs more than once in  the
+        # parent element, which is <iati-activity>
+
+        # FIXME: should database relation be changed to OnetoOne?? see #964
+        activity = self.get_model('Activity')
+        if 'Fss' in self.model_store:
+            for fss in self.model_store['Fss']:
+                if fss.activity == activity:
+                    raise ParserError("Activity", "Fss", "must occur no more "
+                                                         "than once.")
+
+        extraction_date = element.attrib.get('extraction-date')
+        priority = element.attrib.get('priority')
+        phaseout_year = element.attrib.get('phaseout-year')
+
+        if not extraction_date:
+            raise RequiredFieldError(
+                "fss",
+                "extraction-date",
+                "required attribute missing"
+            )
+        validated_extraction_date = self.validate_date(extraction_date)
+
+        if not validated_extraction_date:
+            raise FieldValidationError(
+                "fss",
+                "extraction-date",
+                "extraction-date not of type xsd:date",
+                None,
+                None,
+                element.attrib.get('extraction-date'))
+
+        # phaseout_year must be of type xsd:decimal.
+
+        if phaseout_year is not None:  # 'phasoutout_year' is an optional
+            # attribute.
+            # FIXME: should 'phaseout_year' field be changed to
+            #  DecimalField ?? see #968
+
+            regex = re.compile(r'^((-|\+)?\d*\.?)?\d+$')
```

Comment:
Why do we need regular expressions here? They are very ugly to work with / nobody knows how to read them lol. Correct me if I'm wrong, but I think if you want to check if a value is Decimal, you can just do something like this:

```
from decimal import Decimal

try:
    Decimal(phaseout_year)
except InvalidOperation:
    raise Exception('Number is not decimal')
```

Using try / except block really is a bad practise, but sometimes there's no other way